app/services/reports/partner_info_report_service.rb

```diff
@@ -1,36 +1,43 @@
 module Reports
   class PartnerInfoReportService
+    include ActionView::Helpers::NumberHelper
     attr_reader :year, :organization
-    delegate :partners, to: :organization
-    delegate :count, to: :partners, prefix: true
 
+    # @param year [Integer]
+    # @param organization [Organization]
     def initialize(year:, organization:)
       @year = year
       @organization = organization
     end
 
+    # @return [Hash]
     def report
-      @report ||= {
-        partners_count: partners_count,
-        partner_agency_type: partner_agency_type,
-        partner_zipcodes_serviced: partner_zipcodes_serviced
-      }
+      return @report if @report
+
+      entries = { 'Number of Partner Agencies' => partner_agencies }
+      partner_agency_counts.each do |agency, count|
+        entries["Agency Type: #{agency || 'Unknown'}"] = count
+      end
+      entries['Zip Codes Served'] = partner_zipcodes_serviced
+      @report = { name: 'Partner Agencies and Service Area', entries: entries }
     end
 
-    def columns_for_csv
-      %i[partners_count partner_agency_type partner_zipcodes_serviced]
+    # @return [Array<Partners::Partner>]
+    def partner_agency_profiles
+      @partner_agency_profiles ||= organization.partners.map(&:profile).compact
     end
 
-    def partner_agency_type
-      partners.map do |partner|
-        partner.profile.agency_type
-      end
+    # @return [Integer]
+    def partner_agencies
+      partner_agency_profiles.size
+    end
+
+    def partner_agency_counts
+      partner_agency_profiles.map(&:agency_type).tally
     end
 
     def partner_zipcodes_serviced
-      partners.map do |partner|
-        partner.profile.zips_served
-      end
+      partner_agency_profiles.map(&:zips_served).join(', ')
```

Comment:
Do you need to all .uniq on this? I imagine there is probably a lot of overlap on the zip codes served among partners in a city.

Reply:
Maybe I need more data here. Looking at the local data you sent me, it seems like each partner only has a single zip code. Zip codes also seem to be manually entered and not validated, so users could enter something like "90210 through 90217" or something similar. 😄 I can put in a `uniq`, but I have a feeling the output of this is going to be kind of messy either way.

Reply:
Ah, makes sense. I wasn't aware of the shape of the data.